src/general_tally.cpp

```diff
@@ -0,0 +1,210 @@
+    StaticVector4 all_indexes_(pos_indexes_[iter].indexes_.begin(),
+                               pos_indexes_[iter].indexes_.end());
+    const double d_ = pos_indexes_[iter].distance_in_bin;
+
+    if (energy_in_) {
+      all_indexes_.insert(all_indexes_.begin(), index_E);
+    }
+
+#ifdef ABEILLE_USE_OMP
+#pragma omp atomic
+#endif
+    tally_gen_score(all_indexes_) += d_ * flight_score;
+  }
+}
+
+std::shared_ptr<GeneralTally> make_general_tally(const YAML::Node& node) {
+  // Check the name of the tally is given or not.
+  if (!node["name"]) {
+    fatal_error("No valid name is provided.");
+  }
+  std::string general_tally_name = node["name"].as<std::string>();
+
+  // Check the estimator is given or not.
+  if (!node["estimator"]) {
+    fatal_error(
+        "No, estimator is given for \"" + general_tally_name +
+        "\", therefore, collision estimator will be taken, by default.");
+  }
+  std::string estimator_name = node["estimator"].as<std::string>();
+
+  // Check for the quantity
+  std::string given_quantity = "";
+  if (!node["quantity"]) {
+    fatal_error("No quantity is given for " + general_tally_name + " tally.");
+  }
+  given_quantity = node["quantity"].as<std::string>();
+  GeneralTally::Quantity quant;
+
+  bool found_quantity = false;
+  if (given_quantity == "flux") {
+    quant = GeneralTally::Quantity::Flux;
+    found_quantity = true;
+  }
+
+  if (given_quantity == "fission") {
+    quant = GeneralTally::Quantity::Fission;
+    found_quantity = true;
+  }
+  if (given_quantity == "absorption") {
+    quant = GeneralTally::Quantity::Absorption;
+    found_quantity = true;
+  }
+  if (given_quantity == "elastic") {
+    quant = GeneralTally::Quantity::Elastic;
+    found_quantity = true;
+  }
+
+  if (found_quantity == false) {
+    fatal_error("For " + general_tally_name +
+                " tally, a unknown quantity is given.");
+  }
+
+  // Get the enrgy bounds, if any is given
+  std::shared_ptr<EnergyFilter> energy_filter_ = nullptr;
+  if (node["energy-bounds"]) {
+    std::vector<double> energy_bounds =
+        node["energy-bounds"].as<std::vector<double>>();
+    energy_filter_ = std::make_shared<EnergyFilter>(energy_bounds);
+  }
+
+  // Get the position filter
+  std::shared_ptr<PositionFilter> position_filter_ = make_position_filter(node);
+
+  // For the general tally
+  std::shared_ptr<GeneralTally> itally_genral_tally = nullptr;
+  if (estimator_name == "collision") {
+    itally_genral_tally = std::make_shared<GeneralTally>(
+        position_filter_, energy_filter_, quant,
+        GeneralTally::Estimator::Collision, general_tally_name);
+  }
+
+  if (estimator_name == "track-length")
+    // track-length can be used for flux
+    if (!(given_quantity == "flux")) {
+      fatal_error(
+          "For tally \"" + general_tally_name + "\", the quantity \"" +
+          given_quantity +
+          "\" cannot be evaluated with the selected transport operator.");
+    }
```

Comment:
I don't think this should be here. We have no way of checking here whether a tally is compatible with the transport operator at this point. This is a check which is now done with a helper function on the `ParticleMover` class.